_im-plugin/performance.md

```diff
@@ -0,0 +1,134 @@
+
+If you can afford to increase the amount of time between when a document gets indexed and when it
+becomes visible, increasing the `index.refresh_interval` to a larger value, e.g. 30s, or even disable it in a
+pure indexing scenario, it helps improve indexing speed.
+
+## Index Buffer Size
+
+If the node is doing heavy indexing, be sure this value is large enough. This is set as a percentage of the
+java heap or in bytes. In most cases, the default value of 10% of JVM memory is sufficient2. You can try
+increasing it up to 25% to observe further gains.
+
+## Concurrent merges
+
+Concurrent merges with `max_merge_count`: The concurrentMergeScheduler controls the execution of
+merge operations when they are needed. Merges run in separate threads, and when max number of
+threads is reached, further merges will wait until a merge thread becomes available3.
+In cases where index throttling is an issue, consider increasing the number of merge threads beyond the
+default value.
+
+## Shard distribution
+
+To be sure that the shards are distributed evenly across the data nodes for the index that you're ingesting into, use the following formula to confirm that the shards are evenly distributed:
+
+Number of shards for index = k * (Number of data nodes), where k is the number of shards per node
+
+For example, if there are 24 shards in the index, and there are eight data nodes, then OpenSearch Service assigns three shards to each node. For more information, see [Get started with Amazon OpenSearch Service: How many shards do I need?](https://aws.amazon.com/blogs/database/get-started-with-amazon-elasticsearch-service-how-many-shards-do-i-need/)
+
+## Replica Count to zero
+
+If you anticipate heavy indexing, consider setting the index.number_of_replicas value to "0." Each replica duplicates the indexing process. As a result, disabling the replicas improves your cluster performance. After the heavy indexing is complete, reactivate the replicated indices.
+
+Important: If a node fails while replicas are disabled, you might lose data. Disable the replicas only if you can tolerate data loss for a short duration.
+
+## Experiment to find the optimal bulk request size
+
+Start with the bulk request size of 5 MiB to 15 MiB. Then, slowly increase the request size until the indexing performance stops improving. For more information, see [Using and sizing bulk requests](https://www.elastic.co/guide/en/elasticsearch/guide/2.x/indexing-performance.html#_using_and_sizing_bulk_requests) on the Elasticsearch website.
+
+Note: Some instance types limit bulk requests to 10 MiB. For more information, see [Network limits](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/limits.html#network-limits).
+
+## Use an instance type that has SSD instance store volumes (such as I3)
```

Comment:
Same as above - seems service-related. If so, remove.

Reply:
this is generic. NVME based instances deliver better performance for ingestion. Not specific to opensearch service.